vyper/parser/function_definitions/parse_public_function.py

```diff
@@ -0,0 +1,292 @@
+            context.vars[arg.name] = VariableRecord(arg.name, mem_pos, arg.typ, False)
+        else:
+            context.vars[arg.name] = VariableRecord(
+                arg.name,
+                MemoryPositions.RESERVED_MEMORY + arg.pos,
+                arg.typ,
+                False,
+            )
+
+    # Create "clampers" (input well-formedness checkers)
+    # Return function body
+    if sig.name == '__init__':
+        o = LLLnode.from_list(
+            ['seq'] + clampers + [parse_body(code.body, context)],  # type: ignore
+            pos=getpos(code),
+        )
+    # Is default function.
+    elif sig.is_default_func():
+        if len(sig.args) > 0:
+            raise FunctionDeclarationException(
+                'Default function may not receive any arguments.', code
+            )
+        o = LLLnode.from_list(
+            ['seq'] + clampers + [parse_body(code.body, context)],  # type: ignore
+            pos=getpos(code),
+        )
+    # Is a normal function.
+    else:
+        # Function with default parameters.
+        if sig.total_default_args > 0:
+            function_routine = "{}_{}".format(sig.name, sig.method_id)
+            default_sigs = sig_utils.generate_default_arg_sigs(
+                code, context.sigs, context.global_ctx
+            )
+            sig_chain: List[Any] = ['seq']
+
+            for default_sig in default_sigs:
+                sig_compare, _ = get_sig_statements(default_sig, getpos(code))
+
+                # Populate unset default variables
+                populate_arg_count = len(sig.args) - len(default_sig.args)
+                set_defaults = []
+                if populate_arg_count > 0:
+                    current_sig_arg_names = {x.name for x in default_sig.args}
+                    missing_arg_names = [
+                        arg.arg
+                        for arg
+                        in sig.default_args
+                        if arg.arg not in current_sig_arg_names
+                    ]
+                    for arg_name in missing_arg_names:
+                        value = Expr(sig.default_values[arg_name], context).lll_node
+                        var = context.vars[arg_name]
+                        left = LLLnode.from_list(var.pos, typ=var.typ, location='memory',
+                                                 pos=getpos(code), mutable=var.mutable)
+                        set_defaults.append(make_setter(left, value, 'memory', pos=getpos(code)))
+
+                current_sig_arg_names = {x.name for x in default_sig.args}
+                base_arg_names = {arg.name for arg in sig.base_args}
+                copier_arg_count = len(default_sig.args) - len(sig.base_args)
+                copier_arg_names = current_sig_arg_names - base_arg_names
+
+                # Order copier_arg_names, this is very important.
+                copier_arg_names = {x.name for x in default_sig.args if x.name in copier_arg_names}
```

Comment:
Doesn't putting this in a set strip the ordering?

Reply:
:man_facepalming: yes good find, double checked everything that needs order remain lists now